prometheus-metrics-model/src/main/java/io/prometheus/metrics/model/snapshots/CounterSnapshot.java

```diff
@@ -56,7 +56,17 @@ public CounterDataPointSnapshot(
         Exemplar exemplar,
         long createdTimestampMillis,
         long scrapeTimestampMillis) {
-      super(labels, createdTimestampMillis, scrapeTimestampMillis);
+      this(null, value, labels, exemplar, createdTimestampMillis, scrapeTimestampMillis);
+    }
+
+    public CounterDataPointSnapshot(
```

Comment:
I think builder is enough

Reply:
not sure what you mean here, please elaborate

Reply:
I think we don't need an overloaded ctor - you can just use the builder to pass the metric name